This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <info@kleisauke.nl>
Date: Thu, 16 Aug 2018 17:00:00 +0200
Subject: [PATCH 1/2] Remove build process for eLut.h and toFloat.h

We build these code generators manually.

diff --git a/Half/CMakeLists.txt b/Half/CMakeLists.txt
index 1111111..2222222 100644
--- a/Half/CMakeLists.txt
+++ b/Half/CMakeLists.txt
@@ -1,20 +1,11 @@
 # yue.nicholas@gmail.com
 
 ADD_EXECUTABLE ( eLut eLut.cpp )
-ADD_CUSTOM_COMMAND(
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
-  COMMAND $<TARGET_FILE:eLut> ARGS > ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
-  DEPENDS eLut)
 SET_SOURCE_FILES_PROPERTIES(
   ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
   PROPERTIES HEADER_FILE_ONLY TRUE)
 
 ADD_EXECUTABLE ( toFloat toFloat.cpp )
-ADD_CUSTOM_COMMAND(
-  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
-  COMMAND $<TARGET_FILE:toFloat> ARGS > ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
-  DEPENDS toFloat)
-
 SET_SOURCE_FILES_PROPERTIES(
   ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
   PROPERTIES HEADER_FILE_ONLY TRUE)

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <info@kleisauke.nl>
Date: Thu, 16 Aug 2018 18:00:00 +0200
Subject: [PATCH 2/2] Install pc file on MinGW


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,15 +149,6 @@ CONFIGURE_FILE ( ${ILMBASE_TMP_CONFIG} ${CMAKE_CURRENT_BINARY_DIR}/config/IlmBas
 FILE ( REMOVE ${ILMBASE_TMP_CONFIG} )
 UNSET ( ILMBASE_TMP_CONFIG )
 
-IF ( NOT WIN32 AND OPENEXR_BUILD_SHARED )
-  CREATE_LIBTOOL_FILE ( Half /lib )
-  CREATE_LIBTOOL_FILE ( Iex /lib )
-  CREATE_LIBTOOL_FILE ( IexMath /lib )
-  CREATE_LIBTOOL_FILE ( Imath /lib )
-  CREATE_LIBTOOL_FILE ( IlmThread /lib )
-ENDIF ()
-
-
 # Tests
 IF ( ENABLE_TESTS )
   ADD_SUBDIRECTORY ( HalfTest )
@@ -173,23 +164,24 @@ INSTALL ( FILES
   include/OpenEXR
 )
 
-IF ( NOT WIN32 )
-  FILE ( WRITE ${CMAKE_BINARY_DIR}/IlmBase.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
-  FILE ( APPEND ${CMAKE_BINARY_DIR}/IlmBase.pc "exec_prefix=\${prefix}
+IF ( MINGW OR NOT WIN32 )
+  FILE ( WRITE ${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc "prefix=${CMAKE_INSTALL_PREFIX}\n" )
+  FILE ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc "exec_prefix=\${prefix}
 libdir=\${exec_prefix}/lib
 includedir=\${prefix}/include
-OpenEXR_includedir=\${prefix}/include/OpenEXR
-
+OpenEXR_includedir=\${prefix}/include/OpenEXR\n" )
+  IF (BUILD_ILMBASE_STATIC)
+    FILE ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc "IlmBase_suffix=${ILMBASE_LIBSUFFIX}_s\n" )
+  ELSE ()
+    FILE ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc "IlmBase_suffix=${ILMBASE_LIBSUFFIX}\n" )
+  ENDIF ()
+ FILE ( APPEND ${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc "
 Name: IlmBase
 Description: Base math and exception libraries
 Version: ${ILMBASE_VERSION}
-Libs: -L\${libdir} -lImath${ILMBASE_LIBSUFFIX} -lIexMath${ILMBASE_LIBSUFFIX} -lHalf${ILMBASE_LIBSUFFIX} -lIex${ILMBASE_LIBSUFFIX} -lIlmThread${ILMBASE_LIBSUFFIX} -pthread
-Cflags: -pthread -I\${OpenEXR_includedir}
-")
 
-  INSTALL ( FILES
-    ${CMAKE_BINARY_DIR}/IlmBase.pc
-    DESTINATION
-    lib/pkgconfig
-  )
+Libs.private: -L\${libdir} -lHalf\${IlmBase_suffix} -lIex\${IlmBase_suffix} -lIexMath\${IlmBase_suffix} -lIlmThread\${IlmBase_suffix} -lImath\${IlmBase_suffix} ${CMAKE_THREAD_LIBS_INIT}
+Cflags: ${CMAKE_THREAD_LIBS_INIT} -I\${OpenEXR_includedir}")
+
+  INSTALL ( FILES "${CMAKE_CURRENT_BINARY_DIR}/IlmBase.pc" DESTINATION lib/pkgconfig )
 ENDIF()
